proxy.go

```diff
@@ -61,6 +61,8 @@ type Proxy struct {
 	mitm         *mitm.Config
 	proxyURL     *url.URL
 
+	onClosedConnectionError func(gocontext.Context, string, error)
```

Comment:
we should rename this to something like `onTLSConnectionClosedError`as it called for http connections